### ISSUE TITLE
Auto-corrections

### ISSUE THREAD

**Issue by toydestroyer** [djezzzl/database_consistency#131]
Hey there 👋

I just dropped this wonderful gem into the project that I'm working on, and we have quite a few issues that need to be addressed.

I was wondering if we should have an `--auto-correct` option, similar to the [rubocop](https://docs.rubocop.org/rubocop/usage/auto_correct.html)'s one.

```
database_consistency --auto-correct # auto-correct everything
database_consistency --auto-correct ForeignKeyChecker # auto-correct specific check
database_consistency --auto-correct ForeignKeyChecker app/models/user.rb # auto-correct specific check for a specific model
```

Auto-correction in this case is basically the creation of migration files.

What do you folks think?

I'm happy to contribute if it might be useful to someone else.

**Comment by djezzzl** [comment]
Hi @toydestroyer,

Glad to hear that you liked the gem! 

I like the idea! I can start on it soon. Please let me know if want to contribute (and how much approximately it will take for you) so we don't do the same thing twice.

**Comment by toydestroyer** [comment]
@djezzzl I think I can try to come up with something next week.

Otherwise, if you want to draft the initial idea yourself I will help you implement autocorrections for the rest of the checkers.

**Comment by djezzzl** [comment]
Thank you! Looking forward to see your PR!

**Comment by djezzzl** [comment]
Hi @toydestroyer ,

Could you please share your progress? Maybe you can also open "draft PR" so I can see the direction we are going and maybe help/comment in advance?

**Comment by toydestroyer** [comment]
Hey, @djezzzl I will show something in the next couple of days. I'm currently playing around with auto-corrections for `ForeignKeyChecker`. I will open a draft PR as soon as I have some meaningful structure.

**Comment by djezzzl** [comment]
Hi @toydestroyer ,

Sorry for taking it so long for me to provide the first version.

Just keeping you posted that I've migrated messages to writers [here](https://github.com/djezzzl/database_consistency/pull/135). It was required to continue with auto-correction. Now, we can implement a new writer that will actually correct failures.

**Comment by djezzzl** [comment]
Hi @toydestroyer ,

There is a first auto-fix ready! Please have a look at this PR: https://github.com/djezzzl/database_consistency/pull/136 and let me know if you would like to contribute and implement some of them. 

I would be also happy if you could help me to test them. 

Have a great weekend!

**Comment by djezzzl** [comment]
The PR is in master but I won't release it with a single auto-fix. I would like to have at least a few. It could happen that I will provide more auto-fixes today.

**Comment by djezzzl** [comment]
I have provided a few more auto-fixes. Please check version 1.3.0 and let me know how it works for you if possible. 

Have a great day!

**Comment by toydestroyer** [comment]
Wow! You must've had a very fun weekend 😂

Thanks a lot! I'll play around with it this week and let you know.

**Comment by toydestroyer** [comment]
Bug 1:

If you have multiple foreign keys from one table to another (to the same column) you'll have a duplicate migration name.

Example:

```ruby
class Document
  belongs_to :user
  belongs_to :approved_by, class_name: 'User'
end

class AddUsersDocumentsForeignKey < ActiveRecord::Migration
  def change
    add_foreign_key :documents, :users, column: :user_id, primary_key: :id
  end
end

class AddUsersDocumentsForeignKey < ActiveRecord::Migration
  def change
    add_foreign_key :documents, :users, column: :approved_by_id, primary_key: :id
  end
end
```

Do you want me to open a new issue for this? I can also try to address it myself.

**Comment by djezzzl** [comment]
Wow, thank you a lot! 

The issue you mentioned is fixed here: https://github.com/djezzzl/database_consistency/pull/139. I've released 1.3.1 version.

**Comment by djezzzl** [comment]
Hi @toydestroyer , I've just released version 1.3.5 which has a lot of auto-fixes for most of the migrations. I didn't make any for code manipulation yet, i.e. missing validation. I think I will leave it for a while like that, please feel free to provide a PR if you want to contribute. 

As there are some auto-corrections available, I think I will close the issue. That doesn't mean I will not support the feature anymore. 

Thank you and have a great time!

**Comment by toydestroyer** [comment]
Amazing, thank you so much @djezzzl!

I will be trying this in action in the coming weeks and will let you know.
